internal/handler/comment_test.go

```diff
@@ -0,0 +1,1183 @@
+package handler
+
+import (
+	"bytes"
+	"encoding/json"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+	"time"
+
+	m "github.com/JesusG2000/hexsatisfaction_purchase/internal/handler/mock"
+	"github.com/JesusG2000/hexsatisfaction_purchase/internal/model"
+	"github.com/JesusG2000/hexsatisfaction_purchase/internal/service"
+	"github.com/pkg/errors"
+	testAssert "github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/require"
+	"go.mongodb.org/mongo-driver/bson/primitive"
+)
+
+const (
+	comment             = "comment"
+	text                = "text"
+	slash               = "/"
+	api                 = "api"
+	user                = "user"
+	purchase            = "purchase"
+	period              = "period"
+	authorizationHeader = "Authorization"
+)
+
+func TestComment_Create(t *testing.T) {
+	assert := testAssert.New(t)
+	id := primitive.NewObjectID().Hex()
+	testAPI, err := service.InitTest4Mock()
+	require.NoError(t, err)
+	token, err := testAPI.TokenManager.NewJWT(mock.Anything)
+	require.NoError(t, err)
+
+	type test struct {
+		name    string
+		path    string
+		method  string
+		req     model.CreateCommentRequest
+		fn      func(commentService *m.Comment, data test)
+		expCode int
+		expBody string
+	}
+
+	tt := []test{
+		{
+			name:   "invalid user id",
+			path:   slash + comment + slash + api + slash,
```

Comment:
facepalm )) fmt.Sprintf if your solution here )) And don't use such constants as u used in this file